src/sentry/dynamic_sampling/models/utils.py

```diff
@@ -0,0 +1,62 @@
+from copy import copy
+from dataclasses import dataclass
+from typing import List, Union
+
+from sentry.dynamic_sampling.rules.utils import ProjectId, TransactionName
+
+
+@dataclass
+class DSElement:
+    id: Union[ProjectId, TransactionName]
+    count: float
+    new_count: float = 0.0
+    sample_rate: float = 0.0
+    new_sample_rate: float = 0.0
```

Comment:
I really dislike this.
I understand that for debugging it is fine (and something like this could be used) but I find it 
very confusing to have all these elements in production code.
As far as I'm concerned we only need `id` and anohter `float` (`new_sample_rate`... but if you prefer another derived value it's also fine). 

Also confusing is `sample_rate`, at least for transactions it doesn't make sense to have an independent value for each element.
I don't see it used in the algorithm either so it really puzzles me.

Reply:
in future, we can have sample_rate per project, and this code will be easy to adapt. Right now we need project_id to get the blended_sample rate, not org_id.

Reply:
fixed!